### ISSUE TITLE
ADDED SUBPLOT OF 100 IMAGES

### ISSUE THREAD

**Issue by kevinamexo** [kevinamexo/Machine-Learning#2]
Subplots of the scans as discussed:
https://nbviewer.jupyter.org/urls/dl.dropbox.com/s/t8iq09mx3wq1bpc/COVID_PCA_2.ipynb

**Comment by philusnarh** [comment]
Sample plots of some covid images

![image](https://user-images.githubusercontent.com/8245353/98488338-74466380-2220-11eb-9bc5-7b943a6c4263.png)

Sample plots of some non-covid images
![image](https://user-images.githubusercontent.com/8245353/98488357-8c1de780-2220-11eb-960e-1681908a4130.png)

creating equal dataframe
![image](https://user-images.githubusercontent.com/8245353/98488385-c091a380-2220-11eb-9f2c-2cfcae6564c7.png)

checking the number of components to use;
![image](https://user-images.githubusercontent.com/8245353/98488407-dc954500-2220-11eb-8a1b-82c666b2f5e2.png)

generating a cluster plot of the pcas

![image](https://user-images.githubusercontent.com/8245353/98488436-fc2c6d80-2220-11eb-9b28-e0d3e1df608f.png)

I believe the above cluster plot looks different from yours.

**Comment by kevinamexo** [comment]
Yes please, the plot looks quite different, particularly the COVID
 cluster, which formed a line in my plot.

![index](https://user-images.githubusercontent.com/72938193/98522605-d0cf7000-226c-11eb-94a2-ae5186179f85.png)

**Comment by kevinamexo** [comment]
Please it still gives the same plot after i shuffle the rows before the PCA.

**Comment by kevinamexo** [comment]
Please the PCA is now using 600 images (300 covid, 300 non-covid) , compared to the 790 used previously (395 covid, and 395 non-covid).  That may be why the graph now appears to be missing some points

![image](https://user-images.githubusercontent.com/72938193/98567763-3ab93b00-22a8-11eb-8ae1-1babe222564d.png)

**Comment by kevinamexo** [comment]
I fixed the bug and got this scatterplot:

![image](https://user-images.githubusercontent.com/72938193/98576539-95579480-22b2-11eb-9517-6013a99b2a24.png)

**Comment by philusnarh** [comment]
Looks, promising!

**Comment by philusnarh** [comment]
Okay, now let's combine Kmeans clustering with the PCA: Check this [paper](http://ranger.uta.edu/~chqding/papers/KmeansPCA1.pdf) for more clarification.

**Comment by kevinamexo** [comment]
I combined Kmeans clustering with PCA (using 2 clusters). 

![image](https://user-images.githubusercontent.com/72938193/98938768-4ee77d00-24e0-11eb-8d1b-609f791625ce.png)


I also used performed KMeans clustering without PCA, in this case to return an array containing the cluster each image belongs to.

https://nbviewer.jupyter.org/urls/dl.dropbox.com/s/crew34miycneloj/PCA%2BKMEANS%20COVID.ipynb

**Comment by philusnarh** [comment]
Something on clustering performance metrics:

1. first
<https://medium.com/@ODSC/assessment-metrics-for-clustering-algorithms-4a902e00d92d>

2. second <https://scikit-learn.org/stable/modules/clustering.html>

3. third
<https://www.worldwidejournals.com/indian-journal-of-applied-research-(IJAR)/fileview/August_2014_1406900692__47.pdf>

4. fourth
<https://www.geeksforgeeks.org/ml-v-measure-for-evaluating-clustering-performance/>

**Comment by philusnarh** [comment]
Let's try to produce scatter plots like these too::

![image](https://user-images.githubusercontent.com/8245353/98994221-5f6f1600-2527-11eb-83bf-e42232977571.png)

A scatterplot matrix showing some selected principal components: Let's see the 1st 36 components

**Comment by kevinamexo** [comment]
Alright, I will start working on them

**Comment by kevinamexo** [comment]
Comparison of different Clustering Models  and their V-Measure  (comparing v-measure for different number of clusters), in the case where the data comes in unlabelled.

![image](https://user-images.githubusercontent.com/72938193/99066778-689cc900-25a1-11eb-84a3-65cf2dae433d.png)


1 is a perfect score so it shows that going with 2 clusters is the best choice

**Comment by kevinamexo** [comment]
I compared the various metrics with for the different models. 2 clusters was the best choice so the final part of the code calculates  the metrics for the 2-cluster model

https://nbviewer.jupyter.org/urls/dl.dropbox.com/s/crew34miycneloj/PCA%2BKMEANS%20COVID.ipynb

**Comment by kevinamexo** [comment]
Scatterplot matrix of the first 5 Principal Components

![image](https://user-images.githubusercontent.com/72938193/99406817-45965000-28e6-11eb-9999-52409cbd90a5.png)

**Comment by philusnarh** [comment]
Hi @kevinamexo, try this [package](https://pypi.org/project/med2image/0.2/) and let's see!!!

Alright, will do

**Comment by philusnarh** [comment]
https://stackoverflow.com/questions/38250710/how-to-split-data-into-3-sets-train-validation-and-test/38251213

**Comment by kevinamexo** [comment]
Used the new images for the training:

https://nbviewer.jupyter.org/urls/dl.dropbox.com/s/6p91y3m8taoiy7n/Covid%20Transfer%20Learning.ipynb

**Comment by philusnarh** [comment]
What can you say about your model loss plots?

**Comment by kevinamexo** [comment]
The learning rate too low since it appears almost linear, for the train loss. I think this means it requires more epochs

For the model accuracy plot there is strong over-fitting, because of the gap between the training and validation lines.

EDIT: The validation loss is much higher than the training loss. The model is able to learn the data it was trained on but is not able to generalize, to classify samples outside of the training set. 
There is strong over-fitting. It can be fixed by getting more data(maybe from data augmentation) or sometimes by reducing the complexity of the model

**Comment by philusnarh** [comment]
Not really, your validation plot is wey above the train plot. This means we
are over-fitting.

Moreover, I was expecting your dataset to have normal, pneumonia and
covi-19 images, but this is not so. Any comments on these?

**Comment by kevinamexo** [comment]
I have now added the pneumonia images. I'm now waiting training to end, to see the results

**Comment by philusnarh** [comment]
Okay!

**Comment by kevinamexo** [comment]
https://nbviewer.jupyter.org/urls/dl.dropbox.com/s/sf1fqv6fdjugwmy/load_model.ipynb


Output of filter 100 on 100 different images 
![image](https://user-images.githubusercontent.com/72938193/102484919-d46dc800-405e-11eb-87c9-0a1ce4ec4568.png)

and filter 90:
![image](https://user-images.githubusercontent.com/72938193/102485196-2dd5f700-405f-11eb-8f1f-3b45b80c1f68.png)

**Comment by kevinamexo** [comment]
Here are the dropbox links to the files: 

**Model training:** https://www.dropbox.com/s/6p91y3m8taoiy7n/Model%20Tranining-%20Lung%20Disease%20Classification.ipynb?dl=0

**Model weights h5 file**: https://www.dropbox.com/s/92qn78aakr4mwp8/new_model_weights.h5?dl=0

**Clustering methods on layer activations**:  https://www.dropbox.com/s/jc71yzjnmkqfc7n/Model%20Tranining-%20Lung%20Diseases.ipynb?dl=0

**Comment by johannesa** [comment]
Thanks Kevin.

Johannes

**Comment by kevinamexo** [comment]
**Tasks:**

**1. Show the optimal amount of principal components- Scree plot
2. Scatterplot matrix of the first 10 principal components**
3. Add axis labels to all plots
**4 Plot the layer activations**
**5 Clustering metrics for the layers**

**Comment by kevinamexo** [comment]
https://nbviewer.jupyter.org/urls/dl.dropbox.com/s/gnp729th17puov8/FINAL%20load_model.ipynb

**Comment by kevinamexo** [comment]
**Tasks - 26/05/21**
1. **Write a function to plot and compare the clustering in all the convolution layers**
2. ResNET, **VGG19, Xception,** Inception, AlexNET
**3. Perfomance metrics**
4. Predict the label using untrained data

**Comment by kevinamexo** [comment]
**LATEST VERSION OF PROJECT FILES:** 

Please I had to move the code to Google Colab because the computer i'm using kept freezing and killing the kernel because of memory issues

https://drive.google.com/drive/folders/1vhOLXxICzDA_5feVg7HdrPKEt5gsbExi?usp=sharing
